src/ripple/app/main/Application.cpp

```diff
@@ -400,6 +402,11 @@ class ApplicationImp
 
         , m_journal (logs_->journal("Application"))
 
+        // PerfLog must be started before any other threads are launched.
```

Comment:
Actually, these counters are just the first things for which PerfLog can be used. I have a custom branch that uses PerfLog for logging arbitrary tracing of function calls anywhere in the code base. So it needs to be the first object to start in Application. It may even be appropriate to put it before m_journal in case that code path needs to be traced. Basically, for counters, PerfLog can start as you suggest. But for adding other features in the future, such as application tracing, it needs to come as early as possible.